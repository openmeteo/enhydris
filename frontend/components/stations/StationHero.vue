<template lang="html">
  <section class="hero">
    <div class="hero-body">
      <div class="container">
        <h4 class="subtitle has-text-">
          {{ getDivision() }} {{ data.altitude }} m,
          {{ data.point }}
        </h4>
        <hr />
        <h1 class="title">{{ data.name }}</h1>
        <h2 class="subtitle">{{ data.copyright_holder }} <br /></h2>
        <h3 v-for="type in data.stype" :key="type.id">
          <b-tag rounded>{{ type.descr }}</b-tag>
        </h3>
        <hr />
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: "StationHero",
  props: {
    data: {
      type: Object,
      default: () => {}
    }
  },
  methods: {
    getDivision: function() {
      // extract Division from water_basin/political_division data
      let data = this.$props.data;
      let division;
      division = data["water_basin"] ? data["water_basin"].short_code : null;
      if (!division) {
        division = data["political_division"]
          ? data["political_division"].short_code
          : null;
      }
      return division ? `${division}, ` : "";
    },
    getObject: function(key) {
      let data = this.$props.data;
      return data[key] ? data[key] : "Unknown";
    }
  }
};
</script>

<style lang="css" scoped></style>
