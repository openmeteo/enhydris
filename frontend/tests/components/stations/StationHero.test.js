import { shallowMount, createLocalVue } from "@vue/test-utils";
import StationHero from "@/components/stations/StationHero";
import Buefy from "buefy";

const localVue = createLocalVue();
localVue.use(Buefy);

const stationDetailMock = {
  id: 1403,
  water_basin: null,
  water_division: {
    id: 505,
    last_modified: null,
    name: "ΗΠΕΙΡΟΣ             ",
    short_name: "ΗΠΕΙΡΟΣ ",
    remarks: "",
    area: null,
    mpoly: null,
    water_basin: null,
    water_division: null,
    political_division: null
  },
  political_division: {
    id: 306,
    last_modified: null,
    name: "ΗΠΕΙΡΟΥ",
    short_name: "ΗΠΕΙΡΟΥ",
    remarks: "",
    area: null,
    mpoly: null,
    code: "",
    water_basin: null,
    water_division: null,
    political_division: null,
    parent: 84
  },
  stype: [
    {
      id: 1,
      last_modified: "2011-06-22T05:21:32.781442Z",
      descr: "Meteorological [Μετεωρολογικός]"
    }
  ],
  overseers: [],
  last_modified: "2015-06-02T10:29:57.718660Z",
  name: "Agios Spiridonas",
  short_name: "",
  remarks:
    "ADCON S06. Installed in the framework of project IRMA ETCP GR-IT 2007-13\n\n---ALT---\n\nADCON S06. Εγκαταστάθηκε στο πλαίσιο του προγράμματος IRMA ETCP GR-IT 2007-13",
  srid: 4326,
  approximate: false,
  altitude: 10.0,
  asrid: null,
  point: "SRID=4326;POINT (20.87591 39.14904)",
  is_automatic: false,
  start_date: null,
  end_date: null,
  copyright_holder:
    "Decentralised Administration of Epirus and Western Macedonia",
  copyright_years: "2015",
  owner: 12
};

describe("StationHero.test.js", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(StationHero, {
      localVue,
      propsData: {
        data: stationDetailMock
      }
    });
  });

  it('has received ["data"] as the data property', () => {
    expect(wrapper.vm.data).toEqual(stationDetailMock);
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
