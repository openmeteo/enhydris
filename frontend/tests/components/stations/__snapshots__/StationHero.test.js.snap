// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationHero.test.js has the expected html structure 1`] = `
<section
  class="hero"
>
  <div
    class="hero-body"
  >
    <div
      class="container"
    >
      <h4
        class="subtitle has-text-"
      >
        
         10 m,
        SRID=4326;POINT (20.87591 39.14904)
      
      </h4>
       
      <hr />
       
      <h1
        class="title"
      >
        Agios Spiridonas
      </h1>
       
      <h2
        class="subtitle"
      >
        Decentralised Administration of Epirus and Western Macedonia 
        <br />
      </h2>
       
      <h3>
        <b-tag-stub
          rounded="true"
          tabstop="true"
        >
          Meteorological [Μετεωρολογικός]
        </b-tag-stub>
      </h3>
       
      <hr />
    </div>
  </div>
</section>
`;
